src/pyggdrasil/tree_inference/_config.py

```diff
@@ -0,0 +1,63 @@
+"""Config classes for MCMC sampler,
+ cell simulation and tree inference, distance calculation."""
+
+from pydantic import BaseModel, validator
+
+
+class MoveProbConfig(BaseModel):
+    """Move probabilities for MCMC sampler."""
+
+    prune_and_reattach: float = 0.1
+    swap_node_labels: float = 0.65
+    swap_subtrees: float = 0.25
+
+    @validator("prune_and_reattach", "swap_node_labels", "swap_subtrees")
+    def move_prob_validator(cls, v):
+        """Probabilities sum to 1."""
+        total = sum(v.values())
+        if total != 1:
+            raise ValueError("Move probabilities must sum to 1")
+        return v
+
+    def id(self) -> str:
+        """String representation of move probabilities."""
+        str_rep = "MPC_" + str(self.prune_and_reattach)
+        str_rep = str_rep + "_" + str(self.swap_node_labels)
+        str_rep = str_rep + "_" + str(self.swap_subtrees)
+        return str_rep
+
+
+class McmcConfig(BaseModel):
+    """Config for MCMC sampler."""
+
+    move_probs: MoveProbConfig
+    fpr: float = 1.24e-06
+    fnr: float = 0.097
+    n_samples: int
+    burn_in: int = 0
+    thinning: int = 1
+
+    @validator("fpr")
```

Comment:
Nice validator! (I also think you may appreciate `confloat` by looking how to implement it by adding ~10 character, rather than a few lines :wink: )

Reply:
@pawel-czyz implemented it now with `confloat` but had to silence `pyright`

I run into issues with `pyright` - it does not recognize `confloat`. Do you know a workaround?

`Illegal type annotation: call expression not allowed (reportGeneralTypeIssues)`

No solution here : https://github.com/pydantic/pydantic/issues/5006